syft/file/classifier.go

```diff
@@ -22,6 +22,15 @@ var DefaultClassifiers = []Classifier{
 			`(?m)(?P<version>{{ .version }}\.[0-9]+[-_a-zA-Z0-9]*)`,
 		},
 	},
+	{
+		Class: "nix-store",
+		FilepathPatterns: []*regexp.Regexp{
+			regexp.MustCompile(`/nix/store/[^-]*-([^-]*)-(?P<version>.*)/$`),
+		},
```

Comment:
You might want to use the regexes from https://github.com/nix-community/go-nix/blob/master/pkg/nixpath/nixpath.go#L19-L26 instead - they're a bit more strict.